subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/DefaultPersistentDirectoryCache.java

```diff
@@ -83,6 +83,10 @@ public boolean requiresInitialization(FileLock lock) {
                 return true;
             }
 
+            if (!propertiesFile.exists()) {
```

Comment:
This is probably nit-picking, but shouldn't we check if it is a file, not just that it exists?

Reply:
Yeah I did think about this...it seems nit-picking.